From 56ab9106d1a8acaa94ae86546a897a9a923e1261 Mon Sep 17 00:00:00 2001
From: Martin Pitt <martinpitt@gnome.org>
Date: Mon, 28 Jan 2013 07:41:36 +0100
Subject: [PATCH] keymap: Add HP HDX 9494NR

Add touchpad and www keys.

https://bugzilla.redhat.com/show_bug.cgi?id=757928
(cherry picked from commit 4dbcb1b3bada9632ca3ea763a8d8cf0ffcbc25d5)
---
 Makefile.am                                     | 1 +
 keymaps/hewlett-packard-hdx9494nr               | 3 +++
 src/udev/keymap/95-keyboard-force-release.rules | 2 +-
 src/udev/keymap/95-keymap.rules                 | 1 +
 4 files changed, 6 insertions(+), 1 deletion(-)
 create mode 100644 keymaps/hewlett-packard-hdx9494nr

diff --git a/Makefile.am b/Makefile.am
index 1413073..c5ff79d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2342,6 +2342,7 @@ dist_udevkeymap_DATA = \
 	keymaps/hewlett-packard-tablet \
 	keymaps/hewlett-packard-tx2 \
 	keymaps/hewlett-packard_elitebook-8440p \
+	keymaps/hewlett-packard-hdx9494nr \
 	keymaps/ibm-thinkpad-usb-keyboard-trackpoint \
 	keymaps/inventec-symphony_6.0_7.0 \
 	keymaps/lenovo-3000 \
diff --git a/keymaps/hewlett-packard-hdx9494nr b/keymaps/hewlett-packard-hdx9494nr
new file mode 100644
index 0000000..f3484e1
--- /dev/null
+++ b/keymaps/hewlett-packard-hdx9494nr
@@ -0,0 +1,3 @@
+0xB2 www # FnF3
+0xD8 f22 # touchpad off
+0xD9 f23 # touchpad on
diff --git a/src/udev/keymap/95-keyboard-force-release.rules b/src/udev/keymap/95-keyboard-force-release.rules
index f97a022..3e33e85 100644
--- a/src/udev/keymap/95-keyboard-force-release.rules
+++ b/src/udev/keymap/95-keyboard-force-release.rules
@@ -42,7 +42,7 @@ ENV{DMI_VENDOR}=="Viooo Corporation", ATTR{[dmi/id]product_name}=="PT17", RUN+="
 # These are all the HP laptops that setup a touchpad toggle key
 ENV{DMI_VENDOR}=="Hewlett-Packard*", ATTR{[dmi/id]product_name}=="*[pP][aA][vV][iI][lL][iI][oO][nN]*", RUN+="keyboard-force-release.sh $devpath hp-other"
 ENV{DMI_VENDOR}=="Hewlett-Packard*", ATTR{[dmi/id]product_name}=="*[tT][xX]2*", RUN+="keyboard-force-release.sh $devpath hp-other"
-ENV{DMI_VENDOR}=="Hewlett-Packard*", ATTR{[dmi/id]product_name}=="*2510p*|*2530p*|HP G60 Notebook PC", RUN+="keyboard-force-release.sh $devpath hp-other"
+ENV{DMI_VENDOR}=="Hewlett-Packard*", ATTR{[dmi/id]product_name}=="*2510p*|*2530p*|HP G60 Notebook PC|HDX9494NR", RUN+="keyboard-force-release.sh $devpath hp-other"
 
 ENV{DMI_VENDOR}=="Zepto", ATTR{[dmi/id]product_name}=="Znote 6615WD", RUN+="keyboard-force-release.sh $devpath common-volume-keys"
 
diff --git a/src/udev/keymap/95-keymap.rules b/src/udev/keymap/95-keymap.rules
index 945b131..a13063e 100644
--- a/src/udev/keymap/95-keymap.rules
+++ b/src/udev/keymap/95-keymap.rules
@@ -98,6 +98,7 @@ ENV{DMI_VENDOR}=="Hewlett-Packard", ATTR{[dmi/id]product_name}=="Presario 2100*"
 ENV{DMI_VENDOR}=="Hewlett-Packard", ATTR{[dmi/id]product_name}=="HP G62 Notebook PC", RUN+="keymap $name 0xB2 www"
 ENV{DMI_VENDOR}=="Hewlett-Packard", ATTR{[dmi/id]product_name}=="HP ProBook*", RUN+="keymap $name 0xF8 rfkill 0xB2 www"
 ENV{DMI_VENDOR}=="Hewlett-Packard", ATTR{[dmi/id]product_name}=="HP EliteBook 8440p", RUN+="keymap $name hewlett-packard_elitebook-8440p"
+ENV{DMI_VENDOR}=="Hewlett-Packard", ATTR{[dmi/id]product_name}=="HDX9494NR", RUN+="keymap $name hewlett-packard-hdx9494nr"
 # HP Pavillion dv6315ea has empty DMI_VENDOR
 ATTR{[dmi/id]board_vendor}=="Quanta", ATTR{[dmi/id]board_name}=="30B7", ATTR{[dmi/id]board_version}=="65.2B", RUN+="keymap $name 0x88 media" # "quick play
 
